lib/legacy/class.js

```diff
@@ -155,6 +155,27 @@ function flattenBehaviors(behaviors, list, exclude) {
   return list;
 }
 
+/**
+ * Copies property descriptors from source to target, overwriting all fields
+ * of any previous descriptor for a property *except* for `value`, which is
+ * merged in from the target if it does not exist on the source.
+ * 
+ * @param {*} target Target properties object
+ * @param {*} source Source properties object
+ */
+function mergeProperties(target, source) {
+  for (const p in source) {
+    const targetInfo = target[p];
+    const value = targetInfo && targetInfo.value;
+    const sourceInfo = source[p];
+    if (sourceInfo.value === undefined && value !== undefined) {
```

Comment:
Why not use an `in` check instead of treating `undefined` specially.

Reply:
Yeah, can do.

Reply:
Done